src/python/gudhi/simplex_tree.pyx

```diff
@@ -639,3 +639,12 @@ cdef class SimplexTree:
             self.thisptr = <intptr_t>(ptr.collapse_edges(nb_iter))
             # Delete old pointer
             del ptr
+
+    def __eq__(self, other):
+        """Simplex tree equality operator using C++ depth first search operator==
+
+        :returns: True if the 2 simplex trees are equal, False otherwise.
+        :rtype: bool
+        """
+        cdef intptr_t other_int_ptr=other.thisptr
```

Comment:
I tried other syntaxes, but this one seems the only one that satisfies cython.

My failing attempts:
```python
        return dereference(self.get_ptr()) == dereference(other.get_ptr())                                             ^
src/python/gudhi/simplex_tree.pyx:650:46: Invalid operand type for '*' (Python object)
```

```python
        return dereference(self.get_ptr()) == dereference(<Simplex_tree_interface_full_featured*>(other.thisptr)) 
src/python/gudhi/simplex_tree.pyx:650:58: Python objects cannot be cast to pointers of primitive types
```

Reply:
`def __eq__(self, other:SimplexTree):` seems to help.

Reply:
As I removed the weighted_points and weight_file part, I was no more in need of this feature. I moved it in #463 as it can be interesting from a user point of view. Your trick did it.